.github/CODEOWNERS

```
@@ -3,15 +3,146 @@
+
+## package /packages/gas-fee-controller
+/packages/gas-fee-controller                @MetaMask/confirmations
+/packages/gas-fee-controller/package.json   @MetaMask/confirmations @MetaMask/wallet-framework-engineers
+/packages/gas-fee-controller/changelog.md   @MetaMask/confirmations @MetaMask/wallet-framework-engineers
+
+## package /packages/json-rpc-engine
+/packages/json-rpc-engine              @MetaMask/wallet-api-platform-engineers @MetaMask/wallet-framework-engineers
+
+## package /packages/json-rpc-middleware-stream
+/packages/json-rpc-middleware-stream   @MetaMask/wallet-api-platform-engineers @MetaMask/wallet-framework-engineers
+
+## package /packages/keyring-controller
+/packages/keyring-controller                     @MetaMask/accounts-engineers
+/packages/keyring-controller/package.json        @MetaMask/accounts-engineers @MetaMask/wallet-framework-engineers
+/packages/keyring-controller/changelog.md        @MetaMask/accounts-engineers @MetaMask/wallet-framework-engineers
+
+## package /packages/logging-controller
+/packages/logging-controller                @MetaMask/confirmations
+/packages/logging-controller/package.json   @MetaMask/confirmations @MetaMask/wallet-framework-engineers
+/packages/logging-controller/changelog.md   @MetaMask/confirmations @MetaMask/wallet-framework-engineers
+
+## package /packages/message-manager
+/packages/message-manager                   @MetaMask/confirmations
+/packages/message-manager/package.json      @MetaMask/confirmations @MetaMask/wallet-framework-engineers
+/packages/message-manager/changelog.md      @MetaMask/confirmations @MetaMask/wallet-framework-engineers
+
+## package /packages/name-controller
+/packages/name-controller                   @MetaMask/confirmations
+/packages/name-controller/package.json      @MetaMask/confirmations @MetaMask/wallet-framework-engineers
+/packages/name-controller/changelog.md      @MetaMask/confirmations @MetaMask/wallet-framework-engineers
+
+## package /packages/network-controller
+/packages/network-controller           @MetaMask/wallet-framework-engineers @MetaMask/metamask-assets
+
+## package /packages/notification-controller
+/packages/notification-controller              @MetaMask/notifications
+/packages/notification-controller/package.json @MetaMask/notifications @MetaMask/wallet-framework-engineers
+/packages/notification-controller/changelog.md @MetaMask/notifications @MetaMask/wallet-framework-engineers
+
+## package /packages/notification-services-controller
+/packages/notification-services-controller              @MetaMask/notifications
+/packages/notification-services-controller/package.json @MetaMask/notifications @MetaMask/wallet-framework-engineers
+/packages/notification-services-controller/changelog.md @MetaMask/notifications @MetaMask/wallet-framework-engineers
+
+## package /packages/permission-controller
+/packages/permission-controller        @MetaMask/wallet-api-platform-engineers @MetaMask/wallet-framework-engineers
```

```suggestion
/packages/permission-controller        @MetaMask/wallet-api-platform-engineers @MetaMask/wallet-framework-engineers @MetaMask/snaps-devs
```
I think it makes sense for Snaps to remain a co-owner of the PermissionController. We expect to make more changes to it in the future and are SMEs on the current implementation.